Lib/typing.py

```diff
@@ -1429,11 +1430,16 @@ def name_by_id(user_id: UserId) -> str:
         num = UserId(5) + 1     # type: int
     """
 
-    def new_type(x):
-        return x
+    NewTypeClass = type(name, (object,), {
+        '__supertype__' : tp,
+        '__name__' : name,
+        '__repr__' : lambda self: f"<{self.__name__}:{tp.__name__}>",
+        '__call__' : staticmethod(lambda x: x)
+    })
```

Comment:
I believe this should work similar to how namedtuple works, right?

```python
>>> Point = namedtuple('Point', ['x', 'y'])
>>> Point.__name__, Point.__qualname__, Point.__module__
('Point', 'Point', '__main__')
>>> Point.__doc__
'Point(x, y)'
>>> repr(Point)
"<class '__main__.Point'>"
```

* for `__qualname__` I'm not sure, but seems to be same as `__name__`, so just simplified?
```python
>>> class A:
...       Point = namedtuple('Point', ['x', 'y'])
... 
>>> A.Point.__qualname__
'Point'
```

* for `__module__` I see [namedtuple](https://github.com/python/cpython/blob/17775ae4633ca79468a1f72e92e13a775f1242ca/Lib/collections/__init__.py#L466-L477) used `sys._get_frame` as others in this file do as well.

Reply:
Sure, I haven't looked into `sys._get_frame` and how fast that would be. I'm not sure if people usually pickle a type itself, but maybe they do expect it to behave like a normal types/classes which are all pickle-able.